test/functional/tests/memory/test_memory_metadata_consumption.py

```diff
@@ -0,0 +1,196 @@
+          for cache usage 100% (with 5% tolerance) where cache mode Write-Back is used.
+        pass_criteria:
+          - Successful CAS creation and cores addition.
+          - Filling all cache space successfully.
+          - Memory consumption within limits.
+    """
+    with TestRun.step("Prepare drives and partitions for core devices."):
+        cache_disk = TestRun.disks["cache"]
+        cache_disk.create_partitions([cache_size])
+        cache_dev = cache_disk.partitions[0]
+
+        core_disk = TestRun.disks["core"]
+        parts = [cache_size / cores_count] * cores_count
+        core_disk.create_partitions(parts)
+        core_devices = core_disk.partitions
+
+    with TestRun.step("Start cache with configuration and add all core devices."):
+        free_mem_before_start = get_mem_available()
+
+        cache = casadm.start_cache(cache_dev, cache_mode_wb, cache_line_size, force=True)
+        Udev.disable()
+        # make sure that I/O will go to the cache
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+        core_list = [cache.add_core(dev) for dev in core_devices]
+
+    with TestRun.step("Fill all cache space - run Fio instance on each of cores"):
+        sync()
+        fio = (
+            Fio()
+            .create_command()
+            .direct()
+            .io_engine(IoEngine.libaio)
+            .size(cache.cache_device.size / len(core_list))
+            .offset_increment(cache.cache_device.size / len(core_list))
+            .read_write(ReadWrite.read)
+            .io_depth(64)
+            .block_size(Size(1, Unit.Blocks4096))
+        )
+        for dev in core_list:
+            fio.add_job(f"job_{dev.core_id}").target(dev.path)
+        fio.run()
+
+        occupancy = cache.get_statistics(percentage_val=True).usage_stats.occupancy
+        TestRun.LOGGER.info(f"Cache occupancy: {occupancy}")
+        if occupancy < (100 - percentage_tolerance):
+            TestRun.LOGGER.warning(f"Cache occupancy is below expectation.")
+
+    with TestRun.step("Measure allocates size of metadata."):
+        used_memory = get_module_mem_footprint(cas_cache_module)
+        TestRun.LOGGER.info(f"Memory allocated: {used_memory}")
+
+    with TestRun.step(
+        "Compare allocated size with cache usage statistics " "and required consumption."
+    ):
+        free_mem_after_start = get_mem_available()
+
+        metadata_max_size = calculate_expected_metadata_footprint(cache.size, cache_line_size)
+        TestRun.LOGGER.info(f"Metadata max size: {metadata_max_size}")
+
+        validate_memory_consumption(cache, metadata_max_size, used_memory)
+
+        memory_delta = free_mem_before_start - free_mem_after_start
+        ram_change_status = is_memory_within_limit(
+            "Measured increment of RAM usage after start", metadata_max_size, memory_delta
+        )
+
+        if not ram_change_status:
+            TestRun.LOGGER.error(f"Memory measurements completed, found problems.")
```

Comment:
Error has been already reported by `is_memory_within_limit()`, so no need to print it one more time